.github/workflows/push-release.yml

```diff
@@ -0,0 +1,469 @@
+name: build-docker-images
+on: 
+  push:
+    branches:
+      - "master"
+      - "release/v*"
+    tags:
+      - "v*"
+env:
+  REPOSITORY_OWNER: ${{ github.repository_owner }}
+  IMAGE: ${{ github.repository_owner }}/rancher
+  IMAGE_AGENT: ${{ github.repository_owner }}/rancher-agent
+  IMAGE_INSTALLER: ${{ github.repository_owner }}/system-agent-installer-rancher
+  CATTLE_KDM_BRANCH: dev-v2.8
+jobs:
+  build-chart:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Environment Variables
+        uses: ./.github/actions/setup-tag-env
+      - name: Install dependencies
+        env:
+          HELM_URL_V3: https://get.helm.sh/helm-v3.13.3-linux-amd64.tar.gz
+          HELM_UNITTEST_VERSION: 0.3.2
+        run: |
+          sudo snap install yq
+          curl ${{ env.HELM_URL_V3 }} | tar xvzf - --strip-components=1 -C /tmp/ && \
+          sudo mv /tmp/helm /usr/bin/helm_v3 && \
+          sudo chmod +x /usr/bin/helm_v3
+          helm_v3 plugin install https://github.com/helm-unittest/helm-unittest.git --version ${{ env.HELM_UNITTEST_VERSION }}; \
+      - name: Build
+        run: ./scripts/chart/build chart
+      - name: Validate
+        run: ./scripts/chart/validate
+      - name: Test
+        run: ./scripts/chart/test
+      - name: Package
+        run: ./scripts/chart/package
+      - name: Upload chart
+        uses: actions/upload-artifact@v4
+        with:
+          name: chart
+          path: ./bin/chart/*
+          if-no-files-found: error
+          retention-days: 1
+          overwrite: true
+  publish-chart:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && contains(github.ref, 'tags/') # Only run on push
+    permissions:
+      contents: read
+      id-token: write
```

Comment:
Where is this permission needed?

Reply:
It is required to get secrets using the EIO action

Reply:
True ^